### PR TITLE
Remove meta.address from GET letters on error

### PR DESCRIPTION
**Pull request by kreek**
`address: null` will no longer be included in the response JSON meta object if EVSS returns an error.

### CODE REVIEW

File: lib/evss/letters/letters_response.rb [hunk 1]
```diff
@@ -19,7 +19,9 @@ def initialize(status, response = nil)
       end
 
       def metadata
-        super().merge(address: address)
+        meta = super
+        meta[:address] = address if ok?
```
Comment (omgitsbillryan) [inline lib/evss/letters/letters_response.rb]: I thought the purpose of `meta` was to encapsulate the response from evss to vets-api, in which case, does `address` belong?

I recognize that this question is a bit out of scope for this PR and probably warrants it's own design discussion.

File: spec/support/schemas/letters.json [hunk 2]
```diff
@@ -26,62 +26,78 @@
...
-          "enum": [
-            "OK",
-            "NOT_AUTHORIZED",
-            "NOT_FOUND",
-            "SERVER_ERROR"
-          ]
+        {
+          "properties": {
+            "status": {
+              "enum": [
```
Comment (omgitsbillryan) [inline spec/support/schemas/letters.json]: This enum is reused in multiple places.  Outside the scope of this PR but it probably warrants it's own file with a singular definition.

File: lib/evss/letters/letters_response.rb (hunk 1 above)
Comment (kreek) [inline lib/evss/letters/letters_response.rb]: 👍  it could move up into data, I'll circle back to it during our tech debt sprint
